templates/cc/msg.go

```diff
@@ -102,7 +102,7 @@ bool Validate(const {{ class . }}& m, pgv::ValidationMsg* err) {
 		{{ range .NonOneOfFields }}
 			{{- render (context .) -}}
 		{{ end -}}
-		{{ range .OneOfs }}
+		{{ range .RealOneOfs }}
```

Comment:
Is this going to cause problems for messages with optional fields with validations?

Reply:
generating gofiles with a message like this works just fine now. 

```proto
message AddEntityEntitlementRequest {
    optional string name= 1 [(validate.rules).string = {
        pattern:   "^[^[0-9]A-Za-z]+( [^[0-9]A-Za-z]+)*$",
        max_bytes: 256,
    }]; /** name */
}
```

the regex fails when expected and if no value is provided, no error is thrown, since it's marked as optional. 

I don't know the ins and outs of generation, especially since that piece comes from the original PR, so I'm not sure what the halo effect is like with this change.